### PR TITLE
HealthChecks.IbmMQ

### PR DESCRIPTION
**Pull request by fglaeser**
This PR adds healthchecks for IbmMQ.

I have created HealthChecks.IbmMQ.csproj, functional test and also some documentation.

Best Regards.

### CODE REVIEW

File: src/HealthChecks.IbmMQ/DependencyInjection/IbmMQHealthCheckBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    /// <returns>The <see cref="IHealthChecksBuilder"/>.</returns>
+    public static IHealthChecksBuilder AddIbmMQManagedConnection(this IHealthChecksBuilder builder, string queueManager, string channel, string connectionInfo, string userName = null, string password = null, string name = default, HealthStatus? failureStatus = default, IEnumerable<string> tags = default, TimeSpan? timeout = default)
+    {
+      return builder.Add(new HealthCheckRegistration(
+          name ?? NAME,
+          new IbmMQHealthCheck(queueManager, BuildProperties(channel, connectionInfo, userName, password)),
+          failureStatus,
+          tags,
+          timeout));
+    }
```
Comment (sungam3r) [inline src/HealthChecks.IbmMQ/DependencyInjection/IbmMQHealthCheckBuilderExtensions.cs]:
```suggestion
    }

```

File: src/HealthChecks.IbmMQ/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    };
+    services
+        .AddHealthChecks()
+        .AddIbmMQ("QM.TEST.01", connectionProperties);
+}
+```
+
+### Using Managed Client Connection
+
+For `MQC.TRANSPORT_MQSERIES_MANAGED` connection you can use the following conveniece extension method where you need to specified the channel and the host(port) information. User and password are optional parameters.
```
Comment (sungam3r) [inline src/HealthChecks.IbmMQ/README.md]:
```suggestion
For `MQC.TRANSPORT_MQSERIES_MANAGED` connection you can use the following conveniece extension method where you need to specify the channel and the host(port) information. User and password are optional parameters.
```

File: test/FunctionalTests/HealthChecks.IbmMQ/IbmMQHealthChecksTests.cs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+          .GetAsync();
+
+      response.StatusCode
+          .Should().Be(HttpStatusCode.ServiceUnavailable);
+    }
+
+    [SkipOnAppVeyor]
+    public async Task be_unhealthy_if_ibmmq_managed_is_unavailable()
+    {
+
```
Comment (sungam3r) [inline test/FunctionalTests/HealthChecks.IbmMQ/IbmMQHealthChecksTests.cs]:
```suggestion
```

File: test/FunctionalTests/HealthChecks.IbmMQ/IbmMQHealthChecksTests.cs [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+          .GetAsync();
+
+      response.StatusCode
+          .Should().Be(HttpStatusCode.ServiceUnavailable);
+    }
+
+    [SkipOnAppVeyor]
+    public async Task be_healthy_if_ibmmq_managed_is_available()
+    {
+
```
Comment (sungam3r) [inline test/FunctionalTests/HealthChecks.IbmMQ/IbmMQHealthChecksTests.cs]:
```suggestion
```

**Comment by CarlosLanderas** [comment]
@fglaeser, Im going to review and merge this PR, could you address the conflicts that appeared after merging previous PR's?

Thanks

**Comment by CarlosLanderas** [comment]
@fglaeser Im testing the healthcheck with docker-compose and is working fine. I'm merging it and adding some checks and minor changes. Thank you very much for your contribution

**Pull request merged**